### ISSUE TITLE
pbsv call using --region, fail to find index for svsig.gz files

### ISSUE THREAD

**Issue by Adifo** [PacificBiosciences/pbbioconda#372]
Hello,

I am trying to call SV from multiple samples at once from previously obtained signatures files (.svsig.gz).
As the genome is complex and calling take a lot of time, I would like to constrain pbsv call (2.4.0) to work on a specific region (-r).

I get this kind of error, looking for an index file :
`>|> 20210211 08:26:49.852 -|- FATAL -|- Run -|- 0x7f612497d740|| -|- pbsv call ERROR: Index doesn't exist, failed to load ../calling/F252_HiFi-pbmm2.svsig.gz.tbi`

I have not seen anything about index on the signatures files in pbsv help and my attempt to use tabix did also fail due with an chromosome order error.

How can I produce those .svsig.gz.tbi file ? Does this feature on pbsv call still works or should I rerun the discover step ?

Thank you

**Comment by armintoepfer** [comment]
Hey!

Can you give that a try? 
```
tabix -c '#' -s 3 -b 4 -e 4 input.svsig.gz
```

**Comment by pb-cdunn** [comment]
Or just

    tabix -s 3 -b 4 -e 4 input.svsig.gz

(`#` is the default comment character.)

I'll try to update docs for this.

**Comment by Adifo** [comment]
Hi !

Thank you for the quick answer. I've tried your tabix command but it still fails, with or without the comment line parameter

```
for f in *.svsig.gz; do tabix -c '#' -s 3 -b 4 -e 4 $f; done
[E::hts_idx_push] Unsorted positions on sequence #1: 15534 followed by 1016
tbx_index_build failed: F252_HiFi-pbmm2.svsig.gz
[E::hts_idx_push] Unsorted positions on sequence #1: 90452 followed by 21178
tbx_index_build failed: F2_HiFi-pbmm2.svsig.gz
[E::hts_idx_push] Unsorted positions on sequence #1: 114393 followed by 22934
tbx_index_build failed: F4_HiFi-pbmm2.svsig.gz
[E::hts_idx_push] Unsorted positions on sequence #1: 14488 followed by 9
tbx_index_build failed: MBS847_HiFi-pbmm2.svsig.gz
```

I am using tabix from htslib package. I think it is the last version.
```
tabix --version
tabix (htslib) 1.11
Copyright (C) 2020 Genome Research Ltd.
```

I have also tried an older version but it gives the same kind of error message, just a bit more precision on the line number at which the expected sort order is not respect.

I have taken a look at it and it corresponds to a shift from sequence line starting with a `-` to ones starting with a `~`.
I am not familiar with svsig format so I cannot tell you what it means.

**Comment by armintoepfer** [comment]
In that case, I recommend just redoing discover on the region of interest.
It's much faster for now. We'll investigate internally and fix whatever is
broken. Thank you

Arnaud Di Franco <notifications@github.com> schrieb am Fr., 12. Feb. 2021,
12:12:

**Comment by Adifo** [comment]
Hi.

I've redone the discover and call part. I've had some ressources issues because of the high amount of signatures files (more than 6000 regions) but I managed to make it works by concatening the .svsig files by sample.

Could you think about allowing a bed file entry for the --region parameter (discover and call) (or maybe add a new one to differenciate from the --regions STR) so as to produce less signatures files and smooth this kind of analysis ?

PS: It is ok on my side and I think you can close the issue if needed.

**Comment by armintoepfer** [comment]
We'll add it to the feature wish list. Thanks for using pbsv and I hope it serves you well!
I will leave it open for now, until we've clarified internally why the tabix call doesn't work properly for you.

**Comment by armintoepfer** [comment]
Fixed it internally. Will be part of the next release. Thank you for reporting!

**Comment by bbimber** [comment]
@armintoepfer I am hitting this issue too, using pbsv 2.8.0. 

I'm trying your tabix command on an svsig.gz file, but am getting lots of warnings: 
```
tabix -s 3 -b 4 -e 4 input.svsig.gz

[W::tbx_parse1] Coordinate <= 0 detected. Did you forget to use the -0 option?
```
Any comments on this?

**Comment by bbimber** [comment]
Some other comments. This:
``` 
zcat input.svsig.gz | grep -v '#' | awk ' { $4 <= 0 } '
```
Does not return any lines, so I'm not sure why tabix is complaining about coordinates <=0.

**Comment by khornick1** [comment]
@bbimber I am getting the exact same error. Did you ever find a workaround for this issue?

**Comment by bbimber** [comment]
It's been so long that I dont specifically remember. I forget if this was the error message that went with it; however, I did figure out that when using '-r', each svsig.gz file needs to include a record from that contig. We added a step to check each contig, along the lines of:

```
tabix -l '23001.svsig.gz' | awk ' $1 == "<CONTIG>" ' | wc -l
```

But again, I'm not sure if this was that. I dont think we introduced any changes beyond the check above.
